packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1165,14 +1168,7 @@ export class ProjectLifecycleManager {
   }
 
   setConfigFilePath (lang: 'ts' | 'js') {
-    const configFilePath = this._configFilePath
-
     this._configFilePath = this._pathToFile(`cypress.config.${lang}`)
-
-    if (configFilePath !== this._configFilePath && this._configWatcher) {
-      this.closeWatcher(this._configWatcher)
```

Comment:
This seems to be important. With your changes, are we closing the watcher when we switch testing types?

Reply:
... when we change config file language? 

No. we are not closing and reopening the watcher anymore.

There used to be 2 watchers that did virtually the same thing but with some discrepancies and conflicts (specially in CI.
- One watcher used to watch the current `cypress.config.js`
- The other watcher used to watch all potential config files 

When renaming `cypress.config.js` into `cypress.config.ts` changes would be dealt with twice and break in circle.

Now we only have one single watcher that looks at all files potential and real.
When a change is detected, that is where cypress changes behavior if it's real or potential. 
This way, no need to klankebang any new watcher and the change is guaranteed to run once.